From 8b700cb76f8ee2a254da3d38ca0cbf67cbe36036 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Mon, 22 Jan 2024 11:50:19 +0100
Subject: [PATCH 0580/1695] dts: stm32: Update over-current protection on
 stm32mp257f-ev1

Update over-current protection on stm32mp257f-ev1:
Set BUCK3, BUCK7, LDO2, LDO4, LDO7, LDO8 in HICUP mode to
handle OCP interruption with an error message.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b622fd470105a736b222cd09fc6e357677d00e6f dts: stm32: Update over-current protection on stm32mp257f-ev1]
Change-Id: Iac64ab7e83d5c1e6a3ae1e5b75cd85190ebf9cb5
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/353275
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 core/arch/arm/dts/stm32mp257f-ev1.dts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index e86bf1d78..dbeb892be 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -117,6 +117,7 @@
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <900000>;
 				regulator-always-on;
+				regulator-over-current-protection;
 			};
 			vddio_pmic: buck4 {
 				regulator-name = "vddio_pmic";
@@ -198,6 +199,7 @@
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
+				regulator-over-current-protection;
 				st,pwrctrl_sel = <1>;
 				st,pwrctrl_enable;
 
@@ -227,6 +229,7 @@
 				regulator-name = "vdd_emmc";
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
+				regulator-over-current-protection;
 				st,pwrctrl_sel = <3>;
 				st,pwrctrl_reset;
 			};
@@ -254,6 +257,7 @@
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
+				regulator-over-current-protection;
 			};
 			vpp_ddr: ldo5 {
 				regulator-name = "vpp_ddr";
@@ -284,6 +288,7 @@
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-always-on;
+				regulator-over-current-protection;
 				st,regulator-bypass-microvolt = <3300000>;
 				st,pwrctrl_sel = <3>;
 				st,pwrctrl_reset;
@@ -292,6 +297,7 @@
 				regulator-name = "vddio_sdcard";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
+				regulator-over-current-protection;
 				st,regulator-bypass-microvolt = <3300000>;
 				st,pwrctrl_sel = <3>;
 				st,pwrctrl_reset;
-- 
2.47.2

